### PR TITLE
chore(*): Update typescript, tslint, and react

### PR DESCRIPTION
**Pull request by jrsquared**
Trying to keep up with a few of the staples.

One thing with this is there is now a warning in the console about proptypes being deprecated (formsy-react uses proptypes). This warning will not show up if you are running react in production mode when you deploy.

### CODE REVIEW

File: app/scripts/modules/core/src/api/api.service.ts [hunk 1]
```diff
@@ -12,16 +12,16 @@ interface DefaultParams {
...
+  one?: (...urls: string[]) => IRequestBuilder;
+  all?: (...urls: string[]) => IRequestBuilder;
+  useCache?: (useCache: boolean) => IRequestBuilder;
+  withParams?: (data: any) => IRequestBuilder;
+  data?: (data: any) => IRequestBuilder;
+  get?: (data: any) => IPromise<any>;
+  getList?: (data: any) => IPromise<any>;
+  post?: (data: any) => IPromise<any>;
+  remove?: (data: any) => IPromise<any>;
+  put?: (data: any) => IPromise<any>;
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/api/api.service.ts]: I like this a lot better

**Comment by anotherchrisberry** [comment]
nice

File: app/scripts/modules/core/src/api/api.service.ts (hunk 1 above)
Comment (jrsquared) [reply app/scripts/modules/core/src/api/api.service.ts]: Because it's actually correct now?

Comment (anotherchrisberry) [reply app/scripts/modules/core/src/api/api.service.ts]: Yeah, I often like correct things better than incorrect things.

**Pull request merged**